internal/manifest/docker_schema2_list.go

```diff
@@ -69,23 +78,55 @@ func (list *Schema2ListPublic) Instance(instanceDigest digest.Digest) (ListUpdat
 
 // UpdateInstances updates the sizes, digests, and media types of the manifests
 // which the list catalogs.
-func (list *Schema2ListPublic) UpdateInstances(updates []ListUpdate) error {
-	if len(updates) != len(list.Manifests) {
-		return fmt.Errorf("incorrect number of update entries passed to Schema2List.UpdateInstances: expected %d, got %d", len(list.Manifests), len(updates))
+func (index *Schema2ListPublic) UpdateInstances(updates []ListUpdate) error {
+	editInstances := []ListEdit{}
+	for i, instance := range updates {
+		editInstance := ListEdit{UpdateOldDigest: index.Manifests[i].Digest, UpdateDigest: instance.Digest, UpdateSize: instance.Size, UpdateMediaType: instance.MediaType, ListOperation: listOpUpdate}
+		editInstances = append(editInstances, editInstance)
```

Comment:
One line per field, in the order of the declaration (i.e. with `ListOperation` going first), please.

(Possibly just
```go
… = append(editInstances, ListEdit{
  …
}
```
without the extra variable.)